core/server/data/migrations/versions/4.4/02-migrate-members-signup-access.js

```diff
@@ -0,0 +1,127 @@
+const logging = require('../../../../../shared/logging');
+const {createTransactionalMigration} = require('../../utils');
+const MembersConfigProvider = require('../../../../services/members/config');
+const settingsCache = require('../../../../services/settings/cache');
+const config = require('../../../../../shared/config');
+
+// members_allow_free_signup was migrated to members_signup_access in 4.3/08-migrate-members-signup-setting
+// using a direct binary migration of true=all and false=invite based on new setting behaviour matching the
+// old setting. That was innacurate and the behaviour is changing to a proper invite-only toggle which
+// requires a more nuanced migration.
+//
+// 'all' - no need to change, matches expected behaviour
+// 'invite' - change to 'all' based on members usage
+// 'none' - no need to change, it's a completely new option that wasn't previously migrated to
+//
+// For the majority of cases 'invite' should actually be 'all'. The true invite-only case is:
+// - stripe not configured
+// - portal configured with no plans
+//
+// When switching from signup access from 'invite' to 'all' we also need to remove the 'free' plan from
+// portal settings to avoid paid-only sites suddenly offering free plan signups
+
+module.exports = createTransactionalMigration(
+    async function up(connection) {
+        const currentSetting = await connection('settings')
+            .where('key', 'members_signup_access')
+            .select('value')
+            .first();
+
+        if (!currentSetting) {
+            logging.info('Skipping update of members_signup_access setting. Does not exist');
+            return;
+        }
+
+        if (currentSetting.value !== 'invite') {
+            logging.info(`Skipping update of members_signup_access setting. Not set to 'invite', keeping current value`);
+            return;
+        }
+
+        const membersConfig = new MembersConfigProvider({
+            settingsCache,
+            config
+        });
+
+        const hasStripe = membersConfig.isStripeConnected();
+
+        if (!hasStripe) {
+            logging.info('Skipping update of members_signup_access setting. Stripe is not configured, staying as invite-only');
+            return;
+        }
+
+        const portalPlansSetting = await connection('settings')
+            .where('key', 'portal_plans')
+            .select('value')
+            .first();
+
+        if (JSON.parse(portalPlansSetting.value).length === 0) {
+            logging.info('Skipping update of members_signup_access setting. Portal configured as invite-only, staying as invite-only');
+            return;
+        }
+
+        logging.info(`Updating members_signup_access setting to 'all'`);
+        await connection('settings')
+            .where('key', 'members_signup_access')
+            .update({value: 'all'});
+
+        const currentPlansSetting = await connection('settings')
```

Comment:
🤔 Why do we need to fetch the portal _plans settings here again, we could use the one already fetched [here](https://github.com/TryGhost/Ghost/pull/12907/files#diff-9a8f70a21f28ddd4a9dd7958c1457734c9e983c1eaf12a09acf3ee5c5fc6498aR52) right  ?

Reply:
Good point 😄 was just an oversight, updating...